### ISSUE TITLE
S3 (COS) multipart upload using `USE_TMP_FILE` does not (over)write the file & does not close the file handle

### ISSUE THREAD

**Issue by NielsKorschinsky** [duckdb/duckdb#14877]
### What happens?

Hi,

* CSV file, 1.7 GB size (compressed gzip)
* Uploading into IBM COS (S3)
* Smaller files (with `=auto`) work without issues.

When using the option `USE_TMP_FILE` either explicitly or via `auto` and having a large-enough file, following issues occur:
1. The file is uploaded within 6 minutes, but then another ~14+ minutes pass where the multi-part upload idles without closing. This is reproduceable.
2.  The multipart upload is not closed.
3. The file is not written, even after duckDB says the statement finished successfully.
4. I do not see any `TMP` file while observing the multipart upload, so the premise of `USE_TMP_FILE` seems not fulfilled
5. if a file was present, it is not overwritten.

important: on smaller files, even with `USE_TMP_FILE` enabled, everything works as expected.

When using `=auto`, sometimes the overwrite works and the idle time is gone. I'm assuming in such cases, the tmp file is not used. I cannot exactly tell why it is sometimes used and sometimes not.

When disabling the tmp file, everything works as expected:
1. the file upload finishes within 6 minutes
2. the multipart upload is closed
3. existing files are overwritten, new files created

### To Reproduce

from example: https://duckdb.org/docs/guides/snippets/create_synthetic_data.html

```python
import duckdb

from duckdb.typing import *
from faker import Faker

fake = Faker()

def random_short_text():
    return fake.text(max_nb_chars=20)

con = duckdb.connect()
con.create_function("random_short_text", random_short_text, [], VARCHAR, type="native", side_effects=True)

res = con.sql("""
CREATE TABLE DUCKDB_TABLE_NAME AS
                 SELECT
                    random_short_text() AS short,
                    random_short_text() AS short
                 FROM generate_series(1, 5) s(i)
                 """)
```

I'm unsure how big it needs to be, probably at least 76.5 MiB, as that is the max size for a single part in COS.
I've used 1.7 GB of real data, but the structure is very similar.

```sql
COPY (
    SELECT ALL
    	TRIM("COL1", concat(' ', chr(9), chr(10), chr(13))) AS "COL1",
	IF(TRIM("COL2", concat(' ', chr(9), chr(10), chr(13))) != '', TRIM("COL2", concat(' ', chr(9), chr(10), chr(13))), null) AS "COL2"
    FROM
        "DUCKDB_TABLE_NAME"
) TO
    's3://--iz-bucket-name--/COS_BIG_TABLE.csv'
    (
        FORMAT CSV,
        HEADER false,
        DELIMITER '|',
        ESCAPE '^',
        QUOTE '"',
        DATEFORMAT '%Y-%m-%d',
        TIMESTAMPFORMAT '%Y-%m-%d %H:%M:%S.%f',
        COMPRESSION 'gzip',
        NULLSTR 'NULL',
		USE_TMP_FILE true,        
       FORCE_QUOTE("COL1","COL2")
    )
```

### OS:

Mac

### DuckDB Version:

v1.1.3

### DuckDB Client:

Python

### Full Name:

Niels Korschinsky

### Affiliation:

IBM

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

No - I cannot easily share my data sets due to their large size

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by NielsKorschinsky** [comment]
I'm now even more confused.
I did run multiple tests. I will try to write down just the confusing parts.
I wanted to test with enabled vs disabled flag, but I forgot to actually commit the disable of the flag. 
Therefore I ran four tests with the flag enabled:

it seems like it is not 100% predictable.

> using tmp-file = true

1. flag enabled, a file exists. Error picture as above, file is not overwritten. Multipart upload not finished
2. re-try of the one above, same result.
3. **Deleted all data and pending multipart uploads**
3. the upload did not complete, no file was created. but a multipart upload is pending. -> no file at all is present
4. (I wanted to disable the flag now, but forgot to commit)
5. re-run: everything works as expected, quick upload,  finished upload (**normal execution?!**)
6. re-run: back to "broken" state:  takes forever, unfinished upload.


I will now actually disable the flag and see if that fixes the upload issue.

**Comment by szarnyasg** [comment]
Thanks for investigating this – please let us know what you find!

**Comment by NielsKorschinsky** [comment]
No problem, also my interest to assist here ;) 
I will continue this within this week and update. Cant promise whether tomorrow or the following days.

**Comment by NielsKorschinsky** [comment]
One thing I can already report for sure is that the `.tmp` suffix is not added, even when explicitly set to true:
```
ibmcloud cos multipart-uploads --bucket dev-bucket-name                                                                                        
OK
Found 5 multipart uploads in bucket 'dev-bucket-name ':

UploadId                               Key                                               Initiated   
01000193-3f4b-4767-2726-2e006ddbddb3   DUCKDB_TABLE_NAME.csv   Nov 18, 2024 at 12:41:58   
01000193-3fb5-0860-e1ba-5d937dba631f   DUCKDB_TABLE_NAME.csv   Nov 18, 2024 at 14:37:29   
01000193-3fdd-6a5a-87d4-75b40eda79d9   DUCKDB_TABLE_NAME.csv   Nov 18, 2024 at 15:21:35   
```

I would expect a `.tmp` suffix according to the code here: https://github.com/duckdb/duckdb/blob/c29c67bb971362cd1e9143305acffebb1bc9bd63/src/execution/physical_plan/plan_copy_to_file.cpp#L17

However, it seems like this is disabled whenever the file is remove: https://github.com/duckdb/duckdb/blob/c29c67bb971362cd1e9143305acffebb1bc9bd63/src/planner/binder/statement/bind_copy.cpp#L156

Therefore all of my tests are unnecessary - temp file is disabled anyways on a remote file.
Please update this in the documentation: https://duckdb.org/docs/sql/statements/copy#copy--to-options
I've spend several hours into investigating.


--> Seems like this problem is caused by another issue.

**Comment by NielsKorschinsky** [comment]
I do not see any other option I could use to test. It seems kind of random whether the upload works or fails.
Could be related to the previous fixes to the file closing mechanic, there were some PRs/Issues I remember:
* https://github.com/duckdb/duckdb/issues/12231
* https://github.com/duckdb/duckdb/pull/9758
* It looks similar to this one closed last year November: https://github.com/duckdb/duckdb/issues/9376

However, I'm now lost and would need support in finding a cause.
To summarize the main error picture is:
* big CSV copy to a S3 location.
* a multipart upload is created, but never finishes
* the last part is uploaded within 6 minutes, but the job continues quite long (10+min) after that with low CPU values
* the file is not created nor overwritten
* Unreliable when - sometimes everything works as expected.

**Comment by NielsKorschinsky** [comment]
Hi @szarnyasg ,
thanks for discussing it internally and all your work with duckdb!

IBM COS [supports a subset of the S3 API](https://cloud.ibm.com/apidocs/cos/cos-compatibility) and should classify as AWS-S3 compatible storage system. Therefore I can assume to the best of my knowledge that this bug is also in place when using AWS S3 directly.

However, it is hard for me to test using a paid version.
The AWS free tier only supports up to 5 GB of storage, that could be just enough to reproduce.
Is this enough for you to start fixing the GZIP multipart upload?
